llvm/include/llvm/Frontend/OpenMP/ConstructDecompositionT.h

```diff
@@ -1153,9 +1158,10 @@ template <typename C, typename H> bool ConstructDecompositionT<C, H>::split() {
   for (const ClauseTy *node : nodes) {
     if (skip(node))
       continue;
-    success =
-        success &&
+    bool result =
         std::visit([&](auto &&s) { return applyClause(s, node); }, node->u);
+    if (!isImplicit(node))
```

Comment:
Does this not have to be checked for `allocate` below?

Reply:
`allocate` is not implied in 5.2 or 6.0 by any combination of directives and clauses.  I think it should be an error until some version of the spec allows it to be implied.